### PR TITLE
CI: clang tidy header guards

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES
- CI: add check for header guards

ENDRELEASENOTES

### CODE REVIEW

File: DDCore/include/XML/UnicodeValues.h [hunk 1]
```diff
@@ -14,6 +14,8 @@
 //
 //==========================================================================
 
+#ifdef UNICODE
```
Comment (andresailer) [inline DDCore/include/XML/UnicodeValues.h:17]: clang complains that UNICODE is not defined, so this header file only makes sense if the one that defines UNICODE is included first

File: DDDigi/include/DDDigi/DigiAttenuator.h [hunk 2]
```diff
@@ -44,7 +45,7 @@ namespace dd4hep {
       /// Default destructor
       virtual ~DigiAttenuator();
       /// Callback to read event attenuator
-      virtual double operator()(const DigiCellData& data)  const  override {
```
Comment (andresailer) [inline DDDigi/include/DDDigi/DigiAttenuator.h:47]: clang complained: error: 'operator()' marked 'override' but does not override any member functions

**Comment by andresailer** [comment]
@MarkusFrankATcernch Could you have a look at these changes please? Mostly added some includes for clang to parse the header files without errors. Some comments for special cases.

File: DDCore/include/DD4hep/GrammarUnparsed.h [hunk 3]
```diff
@@ -18,8 +18,8 @@
 // sufficient for all practical purposes.
 //
 //==========================================================================
-#ifndef DD4HEP_DDCORE_GRAMMARUNPARSED_H
-#define DD4HEP_DDCORE_GRAMMARUNPARSED_H
+#ifndef DD4HEP_GRAMMARUNPARSED_H
+#define DD4HEP_GRAMMARUNPARSED_H
 
 #if defined(DD4HEP_DDCORE_GRAMMARPARSED_H)
```
Comment (andresailer) [inline DDCore/include/DD4hep/GrammarUnparsed.h:24]:
```suggestion
#if defined(DD4HEP_GRAMMARPARSED_H)
```

**Comment by MarkusFrankATcernch** [comment]
Thanks a lot for the Sisyphos work to chase up all wrong header guards!

**Comment by andresailer** [comment]
Though the actual continuous Sisyphos work is done by `clang-tidy`

**Pull request merged**